cross-build.sh

```diff
@@ -1,20 +1,29 @@
 #!/usr/bin/env bash
 set -e
 
+DOCKER="docker"
+if ! ${DOCKER} ps >/dev/null 2>&1; then
+    DOCKER="sudo docker"
+fi
+if ! ${DOCKER} ps >/dev/null; then
+    echo "error connecting to docker:"
+    ${DOCKER} ps
+    exit 1
+fi
 # bind last argument in portable manner
 for last in "$@"; do :; done
 
 docker_qemu="docker/qemu-arm-static"
 rm -f ${docker_qemu}
 cp "$(which qemu-arm-static)" ${docker_qemu}
-docker build -t rusty-engine:latest docker/
+${DOCKER} build -t rusty-engine:latest docker/
 rm ${docker_qemu}
 
 if [ "${last}" == "run" ]; then
-    docker run --rm --privileged \
+    cross build --target armv7-unknown-linux-gnueabihf --release
```

Comment:
It suddenly occurs to me that we don't *have* to name the target after a host triple and can make it more readable if we wanted.

I won't do that, but we could do it.